### DISCUSSION TITLE
how i can binding command button in listbox?

### DISCUSSION THREAD

**Discussion by FructusS** [AvaloniaUI/Avalonia#10098] [Q&A] [answered]
Hello, i dont know how i can binding command button in listbox. when i click on the button  nothing happens. I think if i binding incorrectly  button was non clickabled
xaml file 
        <Design.DataContext>
            <vm:MainWindowViewModel x:Name="mainWindowViewModel"/>
        </Design.DataContext>

        <Grid>
            <ListBox Items="{Binding ListTest}" Name="ServiceListBox">
                <ListBox.ItemTemplate>
                    <DataTemplate>
                        <DockPanel>
                            <TextBlock Text="{Binding title}"></TextBlock>
                            <Button Content="delete" 
                                    Command="{Binding ElementName=mainWindowViewModel, 
                                    Path=DeleteServiceCommand}"
                                    CommandParameter="{Binding .}"
                            ></Button>
                        </DockPanel>
                    </DataTemplate>
                </ListBox.ItemTemplate>
            </ListBox>
        </Grid>

viewmodel

    public class MainWindowViewModel : ViewModelBase
    {

    private List<TClass> _test;

    public List<TClass> ListTest
    {
        get => _test;
        set => this.RaiseAndSetIfChanged(ref _test, value);
    }
    public MainWindowViewModel()
    {
        DeleteServiceCommand = ReactiveCommand.Create(dosome);

        ListTest = getlist();
    }
    public ReactiveCommand<Unit, Unit> DeleteServiceCommand { get; }

    private List<TClass> getlist()
    {
        return new List<TClass>
        {
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" },
            new() { title = "123" }
        };
    }
    void dosome()
    {
        var messageBoxStandardWindow = MessageBox.Avalonia.MessageBoxManager
            .GetMessageBoxStandardWindow("title", "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed...");
        messageBoxStandardWindow.Show();
    }
}

**Comment by ahopper** [comment]
Design.DataContext is only used by the previewer, it does not exist at runtime. You could bind to $parent[ListBox].DataContext

**Reply by FructusS** [reply]
can you show example code?i dont get it

**Comment by ahopper** [comment, marked as answer]
Command="{Binding $parent[ListBox].DataContext.DeleteServiceCommand}"
